### DISCUSSION TITLE
How to inform users of moderation actions?

### DISCUSSION THREAD

**Discussion by m88youngling** [LBPUnion/ProjectLighthouse#617] [Ideas]
So, now that we have the new registration system, I've been reminded of something that has been bugging me for some time: how will administrators and moderators communicate moderation actions with users?

For smaller instances, perhaps they could just use off platform communications like Discord. However, for public instances, I'm assuming that some users won't use the same off-site platform like discord. I assumed perhaps instance owners could use email to do this. They could either directly email users about moderation actions or maybe someday we might have some kind of transactional email system that informs you of case information.

However, the new registration system makes it so that theoretically you can play on the server without a verified email. The user may never even see the website! There are two approaches I can see being used to deal with this:

1. We could force users to verify their email, perhaps by giving them a custom announcement screen with instructions but blocking their access to the rest of the server.

2. We could find another way to communicate moderation actions in-game, perhaps via personalized announcement screens. A moderation announcements screen would take priority over the default announcements screen.

3. Users could see specific level deletion / comment deletion cases in a case log on the website on their profile. These would be private to each user. I think this should probably be implemented someday regardless of what solution is picked. However, if users are banned they won't be able to view the case log, so that's a disadvantage. 

I prefer option 1, and both options require additional engineering. Without a solution to communicate moderation action we run the risk of users having no idea what happened to their deleted levels. As you can imagine, this could be bad for a larger instance with a lot of users without verified emails! 

To play the other side of the issue, here is one counterpoint I can think of: If the user has trouble connecting to the server, they *should* already be aware that the website exists since they went through the trouble of patching. If they suddenly lose access to the server, they'll have to go looking for the website where they may be able to get more information about moderation cases. 

Curious to hear people's thoughts on this. Regardless of how you run your instance, I think it's important your users know how and when they are being moderated.

**Comment by sudokoko** [comment]
For bans specifically, perhaps utilizing the custom announcement screen, then disconnecting them from the server? Just a thought branched off of what you've said.

**Comment by m88youngling** [comment]
Mike on Discord suggested that forcing users to have an email address on their account may be a compliance issue with GDPR if they don't have a way to opt out 😬...

**Reply by Ezoiar** [reply]
man i love the eu

**Reply by sudokoko** [reply]
Method I suggested would be great then, combining the announcements feature and disconnecting the user after acknowledgement to display ban information would do great. And just having a custom announcement for any new moderation actions would work.

**Comment by indo2042** [comment]
The easiest solution to this problem, that will technically comply with GDPR, is to avoid using emails entirely. Lighthouse is already capable of sending moderation notifications to users, such as a ban notice.

**Reply by m88youngling** [reply]
The question is, what about users who may never create credentials on the website? They need an email to do this. If they never create their credentials, they can't see website notices. We'd need an ingame solution like a personalized announcements screen

**Reply by indo2042** [reply]
Aren’t users forced to create their account credentials in order to authenticate? Or is their PSN/RPCN linked to their account upon its creation

**Reply by m88youngling** [reply]
How I understand it is that the user connects to the server and is allowed to play and an account is created for them. To interact with website features, they need to create credentials. Without those credentials they can only use the server through the game.

Edit: also, linking your account on the website involves a single authentication request. This is the final step in verifying the account
